src/routes/auth/index.ts

```diff
@@ -1,140 +1,286 @@
-		token,
-		parsedBody.email,
-		{ expirationTtl: 60 * 10 }
-	);
-
-	const activationUrl = `${context.env.BASE_URL}/auth/activate?token=${token}`;
-	const logoUrl = `${context.env.BASE_URL}/assets/torontojs-logo.png`;
-	const emailText = `Please confirm your account by clicking the following link: ${activationUrl}`;
-	const emailHtmlTemplate = generateEmailHtml(activationUrl, logoUrl);
-	const msg = {
-		to: parsedBody.email,
-		from: context.env.SENDER_EMAIL,
-		subject: '[TorontoJS] Confirm your account',
-		text: emailText,
-		html: emailHtmlTemplate
-	};
-	await sgMail.send(msg);
-
-	return context.json<StatusResponse>({ message: 'Created a new profile and sent an email for confirmation' }, StatusCodes.OKAY);
-});
+		const emailExists = await checkEmail(context.env.database, parsedBody.email);
+		if (emailExists) {
+			return context.json({ message: 'Duplicate email' } satisfies StatusResponse, StatusCodes.CONFLICT);
+		}
 
-authRoutes.get('/activate', async (context: Context<EnvironmentBindings>) => {
-	const token = context.req.query('token');
-	if (!token) {
-		return context.json({ message: 'Invalid or missing token' }, StatusCodes.BAD_REQUEST);
-	}
+		const hashedPasswordWithSalt = await hashPassword(parsedBody.password);
 
-	const email = await context.env.ACTIVATION_TOKENS.get(token);
-	if (!email) {
-		return context.json({ message: 'Invalid or expired token' }, StatusCodes.UNAUTHORIZED);
-	}
+		// Store the hashed password and salt in the database
+		parsedBody.password = hashedPasswordWithSalt;
+		await insertProfile(context.env.database, parsedBody);
+
+		// Send email
+		sgMail.setApiKey(context.env.SENDGRID_API_KEY);
+
+		const token = crypto.randomUUID();
+		await context.env.ACTIVATION_TOKENS.put(
+			token,
+			parsedBody.email,
+			{ expirationTtl: 60 * 10 }
+		);
 
-	const emailExists = await checkEmail(context.env.database, email);
-	if (!emailExists) {
-		return context.json({ message: 'User not found' }, StatusCodes.NOT_FOUND);
+		const activationUrl = `${context.env.BASE_URL}/auth/activate?token=${token}`;
+		const logoUrl = `${context.env.BASE_URL}/assets/torontojs-logo.png`;
+		const emailText = `Please confirm your account by clicking the following link: ${activationUrl}`;
+		const emailHtmlTemplate = generateEmailHtml(activationUrl, logoUrl);
+		const msg = {
+			to: parsedBody.email,
+			from: context.env.SENDER_EMAIL,
+			subject: '[TorontoJS] Confirm your account',
+			text: emailText,
+			html: emailHtmlTemplate
+		};
+		await sgMail.send(msg);
+
+		return context.json({ message: 'Created a new profile and sent an email for confirmation' } satisfies StatusResponse, StatusCodes.OKAY);
 	}
+);
+publicAuthRoutes.openapi(
+	createRoute({
+		method: 'get',
+		path: '/',
+		operationId: 'activate',
+		description: 'Received activation email and clicked on activation link',
+		tags: ['Activate'],
+		request: {
+			body: { content: { 'application/json': { schema: SignUpSchema } }, required: true }
```

Comment:
```suggestion
			params: { content: { 'application/json': { schema: SignUpSchema } }, required: true }
```

We expect a query parameter here, not a request body.